src/Native/Unix/System.Native/pal_networking.cpp

```diff
@@ -570,14 +570,10 @@ static int GetHostByNameHelper(const uint8_t* hostname, hostent** entry)
         hostent* result = reinterpret_cast<hostent*>(buffer);
         char* scratch = reinterpret_cast<char*>(&buffer[sizeof(hostent)]);
 
-        int getHostErrno;
+        int getHostErrno = 0;
         int err = gethostbyname_r(reinterpret_cast<const char*>(hostname), result, scratch, scratchLen, entry, &getHostErrno);
         switch (err)
```

Comment:
The man pages do say the return value will specify ERANGE which is also consistent with other reference implementations I found. A quick test shows that if err is ERANGE then getHostErrno will be NETDB_INTERNAL (not ERANGE)